modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/UuidStreamerApplication.java

```diff
@@ -0,0 +1,153 @@
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteDataStreamer;
+import org.apache.ignite.cache.CacheAtomicityMode;
+import org.apache.ignite.cache.CacheMode;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.internal.ducktest.utils.IgniteAwareApplication;
+
+/**
+ * Loading random uuids in cache.
+ */
+public class UuidStreamerApplication extends IgniteAwareApplication {
+    /** {@inheritDoc} */
+    @Override public void run(JsonNode jNode) {
+        String cacheName = jNode.get("cacheName").asText();
+
+        int dataSize = Optional.ofNullable(jNode.get("dataSize"))
+                .map(JsonNode::asInt)
+                .orElse(1024);
+
+        long iterSize = Optional.ofNullable(jNode.get("iterSize"))
+                .map(JsonNode::asLong)
+                .orElse(-1L);
+
+        CacheConfiguration<UUID, byte[]> cacheCfg = new CacheConfiguration<>(cacheName);
+        cacheCfg.setCacheMode(CacheMode.PARTITIONED);
+        cacheCfg.setBackups(2);
+        cacheCfg.setAtomicityMode(CacheAtomicityMode.TRANSACTIONAL);
+        cacheCfg.setIndexedTypes(UUID.class, byte[].class);
+
+        ignite.getOrCreateCache(cacheCfg);
+
+        long start = System.currentTimeMillis();
+
+        markInitialized();
+
+        workParallel(ignite, cacheName, iterSize, dataSize);
+
+        recordResult("DURATION", System.currentTimeMillis() - start);
+
+        markFinished();
+    }
+
+    /** */
+    private void workParallel(Ignite ignite, String cacheName, long iterSize, int dataSize) {
+        int core = Runtime.getRuntime().availableProcessors() / 2;
+
+        long iterCore = 0 < iterSize ? iterSize / core : iterSize;
+
+        ThreadFactory threadFactory = new ThreadFactoryBuilder()
+                .setNameFormat("UuidDataStreamer-%d")
+                .setDaemon(true)
+                .build();
+
+        ExecutorService executors = Executors.newFixedThreadPool(core, threadFactory);
+        CountDownLatch latch = new CountDownLatch(core);
+
+        for (int i = 0; i < core; i++)
+            executors.submit(new UuidDataStreamer(ignite, cacheName, latch, iterCore, dataSize));
+
+        try {
+            while (!terminated()
+                    && !latch.await(1, TimeUnit.SECONDS));
+        }
+        catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        executors.shutdownNow();
+    }
+
+    /** */
+    private class UuidDataStreamer implements Runnable {
+        /** Ignite. */
+        private final Ignite ignite;
+
+        /** Cache name. */
+        private final String cacheName;
+
+        /** Latch. */
+        private final CountDownLatch latch;
+
+        /** Count. */
+        private long cnt = 0L;
```

Comment:
I suppose that it is better to define cnt in run() method as local variable

Reply:
fixed